src/main/java/at/hannibal2/skyhanni/features/misc/chocolatefactory/ChocolateFactoryInventory.kt

```diff
@@ -0,0 +1,117 @@
+    )
+
+    @SubscribeEvent
+    fun onRenderItemOverlayPost(event: GuiRenderItemEvent.RenderOverlayEvent.GuiRenderItemPost) {
+        if (!ChocolateFactoryApi.isEnabled()) return
+        if (!ChocolateFactoryApi.inChocolateFactory) return
+        if (!config.highlightUpgrades) return
+
+        val item = event.stack ?: return
+        val itemName = item.name
+        if (itemName != ChocolateFactoryApi.bestRabbitUpgrade) return
+
+        event.drawSlotText(event.x + 18, event.y, "§6✦", .8f)
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!ChocolateFactoryApi.isEnabled()) return
+        if (!ChocolateFactoryApi.inChocolateFactory) return
+        if (!config.highlightUpgrades) return
+
+        for (slot in InventoryUtils.getItemsInOpenChest()) {
+            if (slot.slotIndex in ChocolateFactoryApi.upgradeableSlots) {
+                if (slot.slotIndex == ChocolateFactoryApi.bestUpgrade) {
+                    slot highlight LorenzColor.GREEN.addOpacity(200)
+                } else {
+                    slot highlight LorenzColor.GREEN.addOpacity(75)
+                }
+            }
+            if (slot.slotIndex == ChocolateFactoryApi.barnIndex && ChocolateFactoryBarnManager.barnFull) {
+                slot highlight LorenzColor.RED
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onRenderItemTip(event: RenderInventoryItemTipEvent) {
+        if (!ChocolateFactoryApi.isEnabled()) return
+        if (!ChocolateFactoryApi.inChocolateFactory) return
+        if (!config.showStackSizes) return
+
+        val item = event.stack
+        val itemName = item.name.removeColor()
+        val slotNumber = event.slot.slotNumber
+
+        if (slotNumber in ChocolateFactoryApi.rabbitSlots) {
+            rabbitAmountPattern.matchMatcher(itemName) {
+                val rabbitTip = when (val rabbitAmount = group("amount").formatInt()) {
+                    in (0..9) -> "$rabbitAmount"
+                    in (10..74) -> "§a$rabbitAmount"
+                    in (75..124) -> "§9$rabbitAmount"
+                    in (125..174) -> "§5$rabbitAmount"
+                    in (175..199) -> "§6$rabbitAmount"
+                    200 -> "§d$rabbitAmount"
+                    else -> "§c$rabbitAmount"
+                }
+
+                event.stackTip = rabbitTip
+            }
+        }
+        if (slotNumber in ChocolateFactoryApi.otherUpgradeSlots) {
+            upgradeTierPattern.matchMatcher(itemName) {
+                event.stackTip = group("tier").romanToDecimal().toString()
+            }
+        }
+        if (slotNumber == ChocolateFactoryApi.milestoneIndex) {
+            item.getLore().matchFirst(unclaimedRewardsPattern) {
+                event.stackTip = "§c!!!"
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onSlotClick(event: GuiContainerEvent.SlotClickEvent) {
+        if (!LorenzUtils.inSkyBlock) return
```

Comment:
inSkyblock is not needed here since the inChocolateFactory should cover it. (When suggestion is implemented in ChocolateFacoryApi at InventoryFullyOpenedEvent)
x4